File: movie-infrastructures/src/main/java/com/movie/movieinfrastructures/repository/TheaterRepositoryImpl.java
package com.movie.movieinfrastructures.repository;

import com.movie.movieapplication.movie.repository.TheaterRepository;
import com.movie.movieapplication.movie.domain.Theater;
import com.movie.movieapplication.movie.entity.TheaterEntity;
import jakarta.persistence.EntityManager;
import java.util.Optional;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Repository;

@Repository
@RequiredArgsConstructor
public class TheaterRepositoryImpl implements TheaterRepository {

    private final TheaterJpaRepository theaterJpaRepository;

    private final EntityManager entityManager;

    @Override
    public Optional<Theater> findById(Long id) {
        TheaterEntity theaterEntity = entityManager.find(TheaterEntity.class, id);
        return Optional.ofNullable(theaterEntity).map(this::toDomain);
    }

    private Theater toDomain(TheaterEntity entity) {
        return Theater.of(entity.getId(), entity.getName());
    }

}

File: infrastructure/src/main/kotlin/com/hanghe/redis/mysql/theater/TheaterRepositoryImpl.kt
package com.hanghe.redis.mysql.theater

import com.hanghe.redis.theater.TheaterEntity
import org.springframework.stereotype.Repository

@Repository
class TheaterRepositoryImpl(
    private val theaterJpaRepository: TheaterJpaRepository
) : TheaterRepository {

    override fun findAll(): List<TheaterEntity> {
        return theaterJpaRepository.findAll()
    }
}
